engine/src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala

```diff
@@ -887,4 +891,13 @@ class SlickDataAccess(databaseConfig: Config) extends DataAccess {
 
     runTransaction(action)
   }
+
+  private def materializeWorkflowDescriptor(id: WorkflowId, workflowSourceFiles: WorkflowSourceFiles): Future[WorkflowDescriptor] = {
+    import akka.pattern.ask
+    val materializedWfDescriptorFuture = (CromwellServer.validateActor ? MaterializeWorkflowDescriptorActor.MaterializeWorkflow(id, workflowSourceFiles)).mapTo[ValidationResult]
```

Comment:
Ugh impedance mismatch between the synchronously invoked API and async messaging to the materialization actor :frowning:  At a minimum this shouldn't be using an `ask`, but I think we might have to live with the basic ugliness as long as we have a synchronous API talking to an actor.